### DISCUSSION TITLE
Serial communication to multiprotocol-TX

### DISCUSSION THREAD

**Discussion by CoMeister** [pascallanger/DIY-Multiprotocol-TX-Module#592] [Q&A]
Hello,
I'm making a custom radio to convert an old radio to 2.4Ghz. I would like to do that in two PCB. One is the screen and input controller.
The second is the radio system. I would like to do a DIY multiprotocol module with CC2500 only (because I have Futaba and Corona SFHSS receivers).
What is the Serial communication protocol between radio and multiprotocolTX?
In which part of this project I have to look?

Thank you 😃

**Comment by wimalopaan** [comment]
Look at Multiprotocoll.{h,ino}

**Reply by CoMeister** [reply]
Thank you, but I don't understand everything.
It would be nice if a Serial command list exist.

**Reply by pascallanger** [reply]
The single command is detailed here: https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/blob/5afdff847716b4ace6d43d7f743b5008830fa398/Multiprotocol/Multiprotocol.h#L816

**Reply by CoMeister** [reply]
Thank you, it's what I need.

**Comment by CoMeister** [comment]
Hello,
I have read the comments on "MULTI protocol serial definition". I can't make a Serial message to bind a receiver or to send the 16 channels.
Is it possible to have a concrete example of a serial command to bind a Futaba SFHSS receiver (R2006GS) and another one to send the 16 channels.
Thank you for your patience.

**Reply by timdriedger** [reply]
I'm looking to do something similar. Did you ever get this to work?

**Reply by Xenon-2** [reply]
https://github.com/Xenon-2/Futaba-Multiprotocol Give a look at this.

**Reply by DespotJosh** [reply]
Would you be willing to share an example of the structure of the commands you send to the module? Specifically the bind command would be greatly appreciated.

**Reply by Xenon-2** [reply]
Hi. Sure.
Here you have the two strings ( in this case for protocol Dsmx F2). Bind command is issued adding 0x80 to the second byte.

You can find all the details  of the communication protocol on Pascallanger multiprotocol.
DsmX 2F
55 6 30 10 C0 5B 33 36 CC B7 E1 40 0 2 10 80 0 4 20 0 0 8 40 0 2 B0 19 8 
Bind
55 86 30 10 C0 5B 33 36 CC B7 E1 40 0 2 10 80 0 2 20 0 1 8 40 0 2 B0 19 8
